src/LmdbCache.ts

```diff
@@ -100,4 +101,63 @@ export class LmdbCache<V = any> implements SortKeyCache<V> {
   storage<S>(): S {
     return this.db as S;
   }
+
+  async prune(entriesStored: number = 1): Promise<PruneStats> {
+    if (!entriesStored || entriesStored <= 0) {
+      entriesStored = 1
+    }
+
+    const statsBefore = await this.db.transaction(() => {
+      const statsBefore: any = this.db.getStats()
+
+      // Keys are ordered, so one particular contract is referred to by consecutive keys (one or more)
+      let entryContractId = ""
+      let entriesCounter = 0
+      this.db.getKeys({ end: null, reverse: true, snapshot: false })
+        .filter(key => {
+          const [contractId] = key.split("|", 1)
+          if (contractId !== entryContractId) {
+            // New entry
+            entryContractId = contractId
+            entriesCounter = 0
+          }
+          // Subsequent entry
+          entriesCounter += 1
+          return entriesCounter > entriesStored
+        }).forEach(key => {
+          // Remove keys over the specified limit
+          this.db.removeSync(key)
+        })
+
+      return statsBefore
+    })
+
+    // All previous writes have been committed and fully flushed/synced to disk/storage
+    await this.db.flushed
+
+    const statsAfter: any = this.db.getStats()
+
+    return {
+      entriesBefore: statsBefore.entryCount,
+      sizeBefore: statsBefore.mapSize,
+      entriesAfter: statsAfter.entryCount,
+      sizeAfter: statsAfter.mapSize
+    } as PruneStats
+  }
+
+  async forEach(callback: (k: CacheKey, v: any) => Promise<void>) {
```

Comment:
hmm, where is it used? the prune method seems to use standard Array's `forEach` method...

Reply:
It's used in the script for copying lmdb cache to leveldb

Reply:
Can we move it then to the script then..? It looks kinda weird here...